### DISCUSSION TITLE
fatal: detected dubious ownership in repository at

### DISCUSSION THREAD

**Discussion by siamesetween** [community/community#48355] [New to GitHub]
Hello, I have several folders that I cannot push to live anymore. I use github online and IAP to do sudo git pull

fatal: detected dubious ownership in repository at

I have tried running git config --global --add safe.directory on IAP but nothing happens. Hope you guys could help. Thanks.

**Comment by neutrinomuon** [comment]
Hello!!!! 

The error message you are seeing typically indicates an issue with the ownership or permissions of the repository. Here are a few steps you can try to resolve the issue:

- Check the ownership and permissions of the repository on the server. Ensure that the user trying to access the repository has the appropriate permissions to do so.

- Try running the "chown" command to change the ownership of the repository to the user who is trying to access it. For example, you can run the command <pre>sudo chown -R username:groupname /path/to/repository</pre> to change the ownership of the repository.

- If you are still having issues, you can try running the "chmod" command to change the permissions of the repository. For example, you can run the command <pre>sudo chmod -R XXX /path/to/repository</pre> to give users access to the repository (XXX are the corresponding number permissions). But, do it with care cause it can be dangerous to allow any permission

If none of the above steps works, you may need to contact the server administrator or GitHub support for further assistance.

I hope this helps. Feel free to let me know if you have any more questions.

**Reply by Ozgurka** [reply]
Benim Trust wallet I want to authorize

**Reply by Ozgurka** [reply]
Benim Trust wallet I want to authorize

**Reply by ykfq** [reply]
`chmod 777` is a bad idea.

**Reply by neutrinomuon** [reply]
Thanks for pointing this out

**Reply by airtower-luna** [reply]
`sudo git pull` (in the original question) is an *extremely* bad idea, too, and possibly the origin of the ownership issues. Git has no need to run as root, and running *anything* as root that doesn't require it is bad practice (because security issues and other bugs can have much more impact).

**Reply by neutrinomuon** [reply]
Yes, this is also the case! Thanks, @airtower-luna

**Reply by HeRo002** [reply]
My 'repository' is an NTFS partition on my SSD. 

My "fatal: detected dubious ownership in repository" was solved by a small change in the '/etc/fstab' file, which automatically mounts disks at start-up on my Ubuntu 20.04. 

In the file '/etc/fstab' I changed this line:

/dev/disk/by-uuid/4AF15A0435E762B4 /mnt/4AF15A0435E762B4 auto nosuid,nodev,nofail,x-gvfs-show,x-gvfs-name=Data-drive 0 0

Into this:

/dev/disk/by-uuid/4AF15A0435E762B4 /mnt/4AF15A0435E762B4 ntfs-3g nosuid,nodev,nofail,uid=1000,gid=1000,x-gvfs-show,x-gvfs-name=Data-drive 0 0

Thus making myself the owner of all files on the NTFS partition. :-)

**Reply by neutrinomuon** [reply]
Many thanks for the clarification and for sharing your experience. This may help several users :)!

**Reply by tahakinn** [reply]
How can I open that file? These directories do not appear when I open them with notes, please answer this, thank you.

**Reply by HeRo002** [reply]
Try going to https://help.ubuntu.com/community/Fstab and search for "Editing fstab".

**Reply by tahakinn** [reply]
Is there any other way to open this file without installing Ubuntu?

**Reply by airtower-luna** [reply]
@tahakinn, if you're not running some kind of Linux your problem isn't that Linux is mounting a Windows partition with overly broad permissions (because Windows filesystems usually lack a similar file ownership concept). You'll need to figure out what the actual problem is to find a solution.

**Comment by jwvanderbeck** [comment]
> The error message you are seeing typically indicates an issue with the ownership or permissions of the repository

This is the tl;dr I see everywhere I google but what confuses me is I'm getting this on a brand new repo that isn't even connected to anything.  IE files on disk, running `git init` followed but `git add .` gives this error and it makes no sense to me.  I've been using git for decades and this is actually the first time I've ever seen this error.

very confusing.  comparing `ls -la` to `whoami` shows identical users.  My only guess is this has something do with Windows funkyness when dealing with network mapped locations to my NAS? Really no clue.  I would  have preferred to actually fix the problem but in the end I just used the suggested option of adding the folder as a safe folder and moved on.

**Comment by Mohamed-E-Fayed** [comment]
I have just met this exception.
The reason in my case was due to having two runs with the same ID.
Note: I load the id from output directory if exists to resume the run.

**Comment by Ramala27** [comment]
atal: detected dubious ownership in repository at 'D:/angular/bookmyshowticket-ui'
'D:/angular/bookmyshowticket-ui' is owned by:
        (inconvertible) (S-1-5-21-3424741159-1584645772-2253200531-1001)
but the current user is:
        DESKTOP-2RUV08D/shayin (S-1-5-21-3338456923-1500767803-1465808457-1001)    
To add an exception for this directory, call:

        git config --global --add safe.directory D:/angular/bookmyshowticket-ui

**Comment by ghostinhershell** [comment]
Hey there @siamesetween, thanks so much for reaching out in our community 👋🏾

Could you please try using a wildcard to add all folders to safe.directory list using the following command:

`git config --global --add safe.directory '*'`

Hope this help, let me know if additional assistance is needed.
